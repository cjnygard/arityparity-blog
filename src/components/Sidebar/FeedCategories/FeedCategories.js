// @flow strict
import React from 'react';
import kebabCase from 'lodash/kebabCase';
import { Link } from 'gatsby';
import type { Categories } from '../../../../types';
import styles from './FeedCategories.module.scss';

type Props = {
  categories: Categories
};

const FeedCategories = ({ categories }: Props) => (
  <div className={styles['feedcategories']}>
    <h3>Categories</h3>
    <ul className={styles['feedcategories__list']}>
      {categories.map((category) => (
        <li className={styles['feedcategories__list-item']} key={category.fieldValue}>
          <Link to={`/tag/${kebabCase(category.fieldValue)}/`}
            className={styles['feedcategories__list-item-link']}
            activeClassName={styles['feedcategories__list-item-link--active']}
          >
            {category.fieldValue} ({category.totalCount})
          </Link>
        </li>
      ))}
    </ul>
  </div>
);

export default FeedCategories;
