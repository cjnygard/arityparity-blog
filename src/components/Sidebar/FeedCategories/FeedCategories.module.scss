@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.feedcategories {
  &__list {
    list-style: none;
    padding: 0px 0px 0px 8px;

    &-item {
      &-link {
        font-size: $typographic-base-font-size;
        color: $typographic-base-font-color;
        font-weight: normal;
        border: 0;

        &:hover,
        &:focus {
          color: $color-primary;
          border-bottom: 1px solid $color-primary;
        }

        &--active {
          color: $color-base;
          border-bottom: 1px solid $color-base;
        }
      }
    }
  }
}
