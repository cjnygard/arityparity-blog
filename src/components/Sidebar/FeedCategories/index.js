// @flow strict
export { default } from './FeedCategories';
