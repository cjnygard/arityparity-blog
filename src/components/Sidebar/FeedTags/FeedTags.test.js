// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import FeedCategories from './FeedTags';

describe('FeedTags', () => {
  const props = {
    tags: [
      {
        fieldValue: 'test_0',
        slug: '/test_0',
        totalCount: 2
      },
      {
        fieldValue: 'test_1',
        slug: '/test_1',
        totalCount: 2
      }
    ]
  };

  it('renders correctly', () => {
    const tree = renderer.create(<FeedCategories {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
