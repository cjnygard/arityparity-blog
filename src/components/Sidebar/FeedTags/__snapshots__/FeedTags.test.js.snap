// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedTags renders correctly 1`] = `
<div
  className="feedtags"
>
  <h3>
    Tags
  </h3>
  <ul
    className="feedtags__list"
  >
    <li
      className="feedtags__list-item"
    >
      <a
        className="feedtags__list-item-link"
        href="/tag/test-0/"
      >
        test_0
         (
        2
        )
      </a>
    </li>
    <li
      className="feedtags__list-item"
    >
      <a
        className="feedtags__list-item-link"
        href="/tag/test-1/"
      >
        test_1
         (
        2
        )
      </a>
    </li>
  </ul>
</div>
`;
